tests/ops/test_contract.py

```diff
@@ -364,7 +403,7 @@ def test_ubersum_collide_error():
     x = torch.randn(a, c)
     y = torch.randn(b, d)
     z = torch.randn(a, b, c, d)
-    with pytest.raises(ValueError, match='Expected tree-structured iarange nesting'):
+    with pytest.raises(NotImplementedError, match='Expected tree-structured iarange nesting'):
```

Comment:
Would it be correct that `naive_ubersum` would handle these error cases now, albeit in exponential time?

Reply:
That is correct.